src/hooks/login/useWebViewLogin.ts

```diff
@@ -3,19 +3,25 @@ import { useDispatch, useSelector } from 'reduxStore/DappProviderContext';
 import { shouldUseWebViewProviderSelector } from 'reduxStore/selectors';
 import { loginWithNativeAuthToken } from 'services/nativeAuth/helpers/loginWithNativeAuthToken';
 import { getWebviewToken } from 'utils/account/getWebviewToken';
+import { getWindowLocation } from '../../utils';
 
 export function useWebViewLogin() {
   const dispatch = useDispatch();
   const shouldUseWebViewProvider = useSelector(
     shouldUseWebViewProviderSelector
   );
+
+  const { search } = getWindowLocation();
+  const urlSearchParams = new URLSearchParams(search) as any;
+  const searchParams = Object.fromEntries(urlSearchParams);
+
   const token = getWebviewToken();
 
   useEffect(() => {
-    if (!shouldUseWebViewProvider) {
+    if (!shouldUseWebViewProvider || !searchParams?.accessToken) {
```

Comment:
what does this logic do? maybe add a comment above

Reply:
for some reason, each dApp tries to authenticate in `webview` and replace the current `loginMethod` (ex. `iframe`) with `extra` which leads to a wrong dApp behavior